Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -4084,18 +4084,28 @@ Ref<HitTestingTransformState> RenderLayer::createLocalTransformState(RenderLayer
     return transformState.releaseNonNull();
 }
 
-static bool parentLayerIsDOMParent(const RenderLayer& layer)
+bool RenderLayer::ancestorLayerIsDOMParent(const RenderLayer* ancestor) const
 {
-    if (!layer.parent())
+    if (!ancestor)
         return false;
-    if (!layer.renderer().element() || !layer.renderer().element()->parentElementInComposedTree())
-        return false;
-    return layer.parent()->renderer().element() == layer.renderer().element()->parentElementInComposedTree();
+    if (renderer().element() && ancestor->renderer().element() == renderer().element()->parentElementInComposedTree())
+        return true;
+
+    PseudoId parentPseudoId;
+    switch (renderer().style().pseudoElementType()) {
+    case PseudoId::ViewTransitionGroup: parentPseudoId = PseudoId::ViewTransition; break;
+    case PseudoId::ViewTransitionImagePair: parentPseudoId = PseudoId::ViewTransitionGroup; break;
+    case PseudoId::ViewTransitionNew: parentPseudoId = PseudoId::ViewTransitionImagePair; break;
+    case PseudoId::ViewTransitionOld: parentPseudoId = PseudoId::ViewTransitionImagePair; break;
+    default: return false;
+    }
```

Comment:
Can we share this code somehow with https://searchfox.org/wubkat/rev/8d7eceb1f07dd13e4e2195f14d02fc27489a87a9/Source/WebCore/style/StyleTreeResolver.cpp#571-583